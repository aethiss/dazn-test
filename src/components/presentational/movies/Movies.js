import React from 'react'
import { Link } from 'react-router-dom'

import { BASE_IMG_URL_200 } from '../../../constants/movieDbUrls'

const Movies = (props) => {

  const { id, poster_path, name, original_title, vote_average } = props

  return (
    <Link className="link" to={`/movie/${id}`}>
      <div className="movie-image">
        <img src={`${BASE_IMG_URL_200}${poster_path}`} />
      </div>
      <div className="movie-info">
        <span className="movie-title">{name || original_title}</span>
        <span className="movie-rating">Rating : {vote_average}</span>
      </div>
    </Link>
  )
}

export default Movies
