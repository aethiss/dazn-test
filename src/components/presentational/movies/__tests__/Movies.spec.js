import React from 'react'
import { shallow } from 'enzyme'

import Movies from '../Movies'

jest.mock('react-router-dom')

const props = {
  id: 1234,
  poster_path: 'path_image.png',
  name: 'title',
  original_title: 'original title',
  vote_average: 'vote_average'
}

describe('@Movies', () => {
  describe('render', () => {
    it('should render default Movie list detail', () => {
      const component = shallow(<Movies {...props} />)
      expect(component).toMatchSnapshot()
    })
    it('should link/redirect to the Movie full details link', () => {
      const component = shallow(<Movies {...props} />)
      expect(component.find('.link').props().to).toBe('/movie/1234')
    })
  })
})
