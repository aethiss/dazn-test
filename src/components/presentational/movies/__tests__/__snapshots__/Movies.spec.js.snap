// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@Movies render should render default Movie list detail 1`] = `
<Link
  className="link"
  to="/movie/1234"
>
  <div
    className="movie-image"
  >
    <img
      src="http://image.tmdb.org/t/p/w200path_image.png"
    />
  </div>
  <div
    className="movie-info"
  >
    <span
      className="movie-title"
    >
      title
    </span>
    <span
      className="movie-rating"
    >
      Rating : 
      vote_average
    </span>
  </div>
</Link>
`;
